### DISCUSSION TITLE
ring led not working. Just the front two headlights

### DISCUSSION THREAD

**Discussion by crazy-interns** [bitcraze/discussions#1556] [Q&A]
The ring led deck used to work properly before but I am not sure why suddenly it stopped working and only the front 2 headlights are on when we turn on the drone. I have already tried flashing the firmware it still does not work.

**Comment by ToveRumar** [comment]
Hi!
Ok, lets find out whats going wrong here. 
Could you please give me some more information. 

- How was the led deck working for you before, did you use any custom effects?

- When you are re-flashing the FW, are you then flashing from release, i.e not custom FW?

- Have you detached and re-attached the deck before it stopped working, or did it suddenly just stop, while attached? Could you give me more information what happened just before it stopped working?

**Comment by crazy-interns** [comment]
How was the led deck working for you before, did you use any custom effects?

It was working well. We ran some codes to do a led light show and it was
running fine.

When you are re-flashing the FW, are you then flashing from release, i.e
not custom FW?

I did it from release.

Have you detached and re-attached the deck before it stopped working, or
did it suddenly just stop, while attached? Could you give me more
information what happened just before it stopped working?

Yes i did detach and reattach even updated the deck from cfclient. I think
it stopped working maybe after i did bootload for the drone from cfclient .

**Reply by ToveRumar** [reply]
Ok, thank you for your answers. 
I still feel like im not fully understanding the sequence of events. 

Have you looked to see what the log looks like? 
How are you controlling the effects, from the FW, lib or from the Client? 

Interesting to hear that it could have stopped working after you boot-loaded the Crazyflie. 
I don't think it should impact, since the LED-deck does not have any FW on it. If the issue has anything to do with bootloading it would be good to uncover.

**Reply by crazy-interns** [reply]
I controlled the effects from the library by running it in vs code. Not
sure why they stopped working suddenly. Just the 2 front headlights are on.

**Reply by gemenerik** [reply]
Have you checked that the deck is mounted the correct way around?
Are you certain that all the decks used [are compatible](https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/#deck---deck)?

**Comment by crazy-interns** [comment]
Yeah I tried removing and reattaching it as well. It was working fine. The
headlights are on so I can tell the led is attached well. Just that none of
the other lights seem to light up for some reason. Is there any way to
reset or restart the led deck?

**Reply by ToveRumar** [reply]
Hi!

There is no way to reset the led deck since it does not have any firmware. It is just connected to the main cpu by pins. 

Ok is it possible for you to send some picture or video of how it looks? 
Does the deck or LEDs feel very hot to the touch, this could indicate some sort of shorting. 

What effect are you using? Is it some custom effect or a per-configured?

**Reply by crazy-interns** [reply]
no it does not get hot. after removing the led ring then updating the lighthouse deck alone. after it restarted i added the ring led and all the lights turned but the cfclient doesnt sense the ring led although it is on now? all the lights are stable at light blue. is this normal?

**Reply by ToveRumar** [reply]
Hi!
Ok so you also have a lighthouse deck connected. If you take it of, does it work then? 

No this does not sound normal, the lighthouse deck and LED deck are compatible. 

What do you mean by "does not sense". Where can you see that its not sensing it, is it in the log output? Could you send those logs please. 
I also need a picture of your drone with the decks. 

Can you change the effects now when the stable blue lights are on? Is the device working normally except the LEDs, i.e can it fly?

I need as much information as possible to be able to help you with this.
